### DISCUSSION TITLE
Rename `src` to `service` and the invoked `onDone` to `serviceDone` ?

### DISCUSSION THREAD

**Discussion by redbar0n** [statelyai/xstate#3574] [Ideas]
Reading: https://xstate.js.org/docs/guides/communication.html#the-invoke-property

`src - the source of the service to invoke`

I found it a bit weird that `src` was used as the key, when it (afaict) always signifies a `service`, which is more semantically meaningful. Imo, it's like the difference between calling a variable `data` instead of the name of the data it contains.

Also, it makes mentally connecting the part of the machine:
```js
invoke: {
        src: "fetchUsers",
```

...with its usage:

```js
const [state, send] = useMachine(usersMachine, {
  services: {
    fetchUsers: ...
  }
```

a little bit harder. Especially if they are in different files. Since you have to remember that `src` maps to one of the `services`.

I also saw this big yellow `WARNING` mentioned at the [same docs](https://xstate.js.org/docs/guides/communication.html#the-invoke-property):

>Don't get the onDone property on a state confused with invoke.onDone - they are similar transitions, but refer to different things.

```js
loading: {
  invoke: {
    src: someSrc,
    onDone: {/* ... */} // refers to `someSrc` being done
  },
  initial: 'loadFoo',
  states: {
    loadFoo: {/* ... */},
    loadBar: {/* ... */},
    loadingComplete: { type: 'final' }
  },
  onDone: 'loaded' // refers to 'loading.loadingComplete' being reached
}
```

How about a slight API change, so it is self-explanatory and the comments there can be removed?
Rename `src` to `service` and the invoked `onDone` to `serviceDone`.
Like so:

```js
loading: {
  invoke: {
    service: someService,
    serviceDone: {/* ... */}
  },
  initial: 'loadFoo',
  states: {
    loadFoo: {/* ... */},
    loadBar: {/* ... */},
    loadingComplete: { type: 'final' }
  },
  onDone: 'loaded'
}
```

The invoked `onError` could be similarly renamed to `serviceError`.

**Comment by davidkpiano** [comment]
Great suggestions!

To the first one, `services` will actually be renamed to `actors` ("everything is an actor") so this might make more sense:

```js
// FUTURE
invoke: {
  actor: 'someActor'
}
```

**Reply by redbar0n** [reply]
Thanks! That makes a lot of sense!

So  it could be:

```js
invoke: {
  actor: 'someActor'
  actorDone: {},
  actorError: {},
}
```

**Reply by davidkpiano** [reply]
We'd like to keep the `onDone`, `onError`, etc. convention for now because they essentially mean the same thing regardless of if it's a state or invocation: a sub-process is done.

**Reply by redbar0n** [reply]
yeah, but that gives the aforementioned problem: that people, glancing at the long nested JSON definition, will get the them easily mixed up. Because the context to understand them could be far removed.. Better to be more explicit and be able to delete documentation warnings.
